qulice-findbugs/src/main/java/com/qulice/findbugs/FindBugsExcludes.java

```diff
@@ -0,0 +1,199 @@
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.qulice.findbugs;
+
+import com.qulice.spi.Environment;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.dom4j.Document;
+import org.dom4j.DocumentHelper;
+import org.dom4j.Element;
+import org.dom4j.tree.BaseElement;
+
+/**
+ * Encapsulates the exclusion rules.
+ * @since 0.18.14
+ */
+public class FindBugsExcludes {
+
+    /**
+     * Environment.
+     */
+    private final Environment env;
+
+    /**
+     * Ctor.
+     * @param env Environment
+     */
+    public FindBugsExcludes(final Environment env) {
+        this.env = env;
+    }
+
+    /**
+     * Exclusions as FindBugs filter XML file.
+     * @return XML
+     */
+    public final Document asXml() {
+        final Document document = DocumentHelper.createDocument();
+        // @checkstyle LineLengthCheck (1 line)
+        final FindBugsExcludes.ExcludesXml xml = new FindBugsExcludes.ExcludesXml(
+            document.addElement("FindBugsFilter")
+        );
+        this.env.excludes("findbugs").stream().forEach(xml::addExclude);
+        return document;
+    }
+
+    /**
+     * Converts excludes to command argument
+     * as a path to the file.
+     * @return List of arguments (zero or one)
+     */
+    public final Collection<String> asArguments() {
+        final Document document = this.asXml();
+        final Element root = document.getRootElement();
+        final List<String> arguments = new LinkedList<>();
+        if (root.hasContent()) {
+            final File file = this.createTempFile();
+            FindBugsExcludes.writeXml(
+                file,
+                document
+            );
+            arguments.add(file.getAbsolutePath());
+        }
+        return arguments;
+    }
+
+    /**
+     * Writes xml to file.
+     * @param file File the XML will be written to.
+     * @param document The XML document that needs to be saved.
+     */
+    private static void writeXml(final File file, final Document document) {
+        try (final Writer writer = new OutputStreamWriter(
+            new FileOutputStream(file),
+            "UTF-8"
+        )) {
+            document.write(writer);
+        } catch (final IOException exc) {
+            throw new IllegalStateException(
+                "can't generate exclude rules for findbugs",
+                exc
+            );
+        }
+    }
+
+    /**
+     * Creates a temporary file.
+     * @return Temp file
+     */
+    private File createTempFile() {
```

Comment:
@ilyakharlamov Let's use cactoos `TempFile` here

Reply:
@paulodamaso  I tried. Interface `Validator.validate()` throws `ValidationException`, but TempFile throws generic `java.lang.Exception` (why not IOException???).  I must use temp file inside the validate() so I must catch Exceotion, but style rules deprecate catching java.lang.Exception. Decided to use JDK nio instead.